File: Part13.java
import java.util.HashMap;

public class Part13 {
    public static void main(String[] args){
        String s = "III";
        HashMap<String,Integer> m = new HashMap<>();

        m.put("I",1);
        m.put("V",5);
        m.put("X",10);
        m.put("L",50);
        m.put("C",100);
        m.put("D",500);
        m.put("M",1000);
    int result =0;

    for(int i=0; i <s.length();i++){
        if(i < s.length() - 1 && m.get(s.charAt(i)) < m.get(s.charAt(i + 1))){
            result -= m.get(s.charAt(i));
        }else{
            result += m.get(s.charAt(i));
        }
    }
    System.out.println(result);
    }
}

File: Part13.cpp
#include<iostream>
#include <unordered_map>
using namespace std;
int main (){
    string s = "III";
    unordered_map<char,int> m;
    m['I'] = 1;
    m['V'] = 5;
    m['X'] = 10;
    m['L'] = 50;
    m['C'] = 100;
    m['D'] = 500;
    m['M'] = 1000;

    int result = 0;

    for(int i =0; i< s.length();i++){
        if(m[s[i]]<m[s[i+1]]){
            result -= m[s[i]];
        }
        else{
            result += m[s[i]];
        }
    }
    cout << result << endl;
    return 0;
}